examples_tests/35.TAA/velbuf.vert

```diff
@@ -0,0 +1,22 @@
+#version 430 core
+
+layout (location = 0) in vec3 vPos;
+
+layout (location = 0) uniform mat4 MVP;
+layout (location = 1) uniform mat4 prevMVP;
+
+out vec2 vVelocity;
+
+void main()
+{
+	vec4 p_cs = MVP*vec4(vPos, 1.0);
+	gl_Position = p_cs;
+	
+	// convert to ndc
+	p_cs /= p_cs.w;
```

Comment:
nope, Z-divide must happen per pixel (move it to pixel shader and pass `xyw` along).

Reply:
You will have epic arithmetic, divide by 0 and other errors on interpolation for triangles close to the near plane... this is really needed per pixel

Reply:
@Crisspl if you just nuked the Velocity Buffer and went with Deferred+ Ready idea from here
https://github.com/buildaworldnet/IrrlichtBAW/pull/341#issuecomment-518009418

then all of the above would go away and be a non-issue